src/compiler/index.js

```diff
@@ -6,12 +6,16 @@ import {getBuilders} from './builders';
 /**
  * Returns the AST for an AMP Document with eligible components server-rendered.
  *
- * @param {!./types.TreeProtoDef} ast
- * @param {!./types.VersionsDef} versions
- * @return {!./types.TreeProtoDef}
+ * @param {!./types.CompilerRequest} request
+ * @return {!./types.CompilerResponse}
  */
-function compileAst(ast, versions) {
-  return compiler.renderAst(ast, getBuilders(versions));
+function compile(request) {
+  // TODO(samouri): remove the defaults.
+  const document = request.document ?? {root: 0, tree: []};
+  const versions = request.versions ?? {'amp-layout': 'v0'};
+
+  // TODO(samouri): handle type=error case.
```

Comment:
What is `type=error`? Can we just throw the error?

Reply:
Sure! Is that the best way of doing it? Currently `bento-compiler` will return `type: success, value: Tree` or `type: failure, error: string`. Happy to throw if that is the best way to receive the error from c++ side

Reply:
Discussed offline and online